src/promptflow/promptflow/exceptions.py

```diff
@@ -302,18 +314,11 @@ def _error_message(cls, e: Exception):
     @classmethod
     def _error_detail(cls, e: Exception):
         exception_codes = cls._get_exception_codes(e)
-        exception_code = None
-        for item in exception_codes[::-1]:
-            if "promptflow" in item["module"]:  # Only record information within the promptflow package
-                exception_code = item
-                break
-        if not exception_code:
-            return ""
-        return (
-            f"module={exception_code['module']}, "
-            f"code={exception_code['exception_code']}, "
-            f"lineno={exception_code['lineno']}."
-        )
+        promptflow_codes = ""
+        for item in exception_codes:
+            if "promptflow" in item["module"]:  # Only record the promptflow package and code.
+                promptflow_codes += f"{item['module']}, line {item['lineno']}, {item['exception_code']}\n"
+        return promptflow_codes
```

Comment:
The format of returned value seems to be changed greatly, is that be design? And original resuls only return info of most inner code, but new result returns info from all codes.

Reply:
Yes, originally only returning one line of error code, now returning traceback within promptflow helps locate the upstream entry of the error, making it easier to cache the error somewhere upstream.